### ISSUE TITLE
options support by {type} / buffer local options

### ISSUE THREAD

**Issue by promisedlandt** [ludovicchabant/vim-gutentags#264]
For a while, ripper tags (for Ruby) was supported by adding following options like this: `let g:gutentags_ctags_executable_ruby = 'ripper-tags --ignore-unsupported-options --recursive'`

(see [this](https://github.com/ludovicchabant/vim-gutentags/issues/87#issuecomment-558912534) comment).

This does not work anymore, because the executable is now passed as the whole string, and of course there is no executable called `'ripper-tags --ignore-unsupported-options --recursive`.
Options are now passed separately, but setting them as  `let g:gutentags_ctags_extra_args = ['--ignore-unsupported-options']` breaks all non-ripper invocations, because ctags dies with `ctags: Unknown option: --ignore-unsupported-options`.

In other words, the extra options only make sense with the ripper-tags executable.

Would it be possible to add either `g:gutentags_ctags_extra_args_{type}` or use buffer local `b:gutentags_ctags_extra_args` so it's possible to set it in after/filetype/ruby.vim ?

**Comment by smati-checkr** [comment]
+1 Facing the same issue.

Not entirely sure why `g:gutentags_ctags_executable_ruby` functionality was taken away without a robust alternative.

**Comment by ludovicchabant** [comment]
The functionality was taken away without a robust alternative because I just didn't think of it :)

FYI, the change was caused by the giant mess that is whitespace handling in shell scripts, for people who put vim files/projects/gutentags in directories with spaces in them.

So yes you're right we need to add a `g:gutentags_ctags_extra_args_{type}` option.
